tests/php/core-api/test_class.jetpack-core-api-xmlrpc-consumer-endpoint.php

```diff
@@ -0,0 +1,68 @@
+<?php
+require_once JETPACK__PLUGIN_DIR . '/_inc/lib/core-api/class.jetpack-core-api-xmlrpc-consumer-endpoint.php';
+
+class WP_Test_Jetpack_Core_Api_Xmlrpc_Consumer_Endpoint extends WP_UnitTestCase {
+
+	private $home_url = 'http://example.com';
+
+	public function setUp() {
+		parent::setUp();
+	}
+
+	/**
+	 * @author zinigor
+	 * @covers Jetpack_Core_XMLRPC_Consumer_Endpoint
+	 * @requires PHP 5.2
+	 * @dataProvider _true_false_provider
+	 */
+	public function test_Jetpack_Core_API_XMLRPC_Consumer_Endpoint_privacy_check( $query_success, $result ) {
+		Jetpack::load_xml_rpc_client();
+
+		$xmlrpc_mock = $this->getMockBuilder( 'Jetpack_IXR_Client' )
+					 ->setMethods( array( 'query', 'getResponse' ) )
+					 ->getMock();
+
+		$endpoint = new WP_Test_Dummy_Xmlrpc_Consumer_Endpoint( $xmlrpc_mock );
+
+		$xmlrpc_mock->expects( $this->once() )
+			->method( 'query' )
+			->with( 'jetpack.isSitePubliclyAccessible', home_url() )
+			->willReturn( $query_success );
+
+		if ( $query_success ) {
+			$xmlrpc_mock->expects( $this->once() )
+				->method( 'getResponse' )
+				->willReturn( $result );
+		} else {
+			$xmlrpc_mock->expects( $this->never() )
+				->method( 'getResponse' );
+		}
+
+		$this->assertEquals( $result, $endpoint->process( null ) );
+	}
+
+	public function _true_false_provider() {
```

Comment:
I think I made it so because of the habit of making private and protected method names start with an underscore. But later PHPUnit must have barfed on that and I had to change the method to public, forgetting to remove the underscore.